### PR TITLE
fix: inject oauth params when generating sync input

### PR DESCRIPTION
**Pull request by pedroslopez**
## What

When moving the configuration fetching to happen at the attempt level in https://github.com/airbytehq/airbyte/pull/21629 we stopped injecting OAuth parameters into the configuration. This led to errors due to not having the right params during syncs. 

## How

- Properly inject oauth prams.

### CODE REVIEW

**Review by alafanechere — APPROVED** [review]

**Comment by mfsiega-airbyte** [comment]
(Left a comment along these lines in Slack, but for visibility)

What testing have we done to make sure this is safe?

**Comment by pedroslopez** [comment]
> (Left a comment along these lines in Slack, but for visibility)
> 
> What testing have we done to make sure this is safe?

@mfsiega-airbyte I've added tests here to verify that we're injecting oauth params, but currently deploying this to dev-2 to check it out in cloud. Will merge once I confirm there.

**Comment by mfsiega-airbyte** [comment]
> @mfsiega-airbyte I've added tests here to verify that we're injecting oauth params, but currently deploying this to dev-2 to check it out in cloud. Will merge once I confirm there.

Did we not do this for the original PR?

**Comment by pedroslopez** [comment]
> > @mfsiega-airbyte I've added tests here to verify that we're injecting oauth params, but currently deploying this to dev-2 to check it out in cloud. Will merge once I confirm there.
> 
> Did we not do this for the original PR?

I did not do this with an OAuth connector for the original PR.

**Review by evantahler — APPROVED** [review]

File: airbyte-workers/src/test/java/io/airbyte/workers/temporal/scheduling/activities/GenerateInputActivityTest.java [hunk 1]
```diff
@@ -225,6 +239,8 @@ void testGetResetSyncWorkflowInput() throws IOException, ApiException {
         .withDestinationConfiguration(DESTINATION_CONFIGURATION)
         .withState(STATE);
 
+    verify(oAuthConfigSupplier).injectDestinationOAuthParameters(DESTINATION_DEFINITION_ID, WORKSPACE_ID, DESTINATION_CONFIGURATION);
```
Comment (evantahler) [inline airbyte-workers/src/test/java/io/airbyte/workers/temporal/scheduling/activities/GenerateInputActivityTest.java:242]: nit: could this test ensure that a secret was decrypted?

Comment (pedroslopez) [reply airbyte-workers/src/test/java/io/airbyte/workers/temporal/scheduling/activities/GenerateInputActivityTest.java:245]: Updated the test to actually check that we're using the config returned by the injection method. Internals of how the injection works left up to unit tests for the OAuthConfigSupplier.

**Pull request merged**